Faced unexpected panicking on types with Resolve implementation.

I faced quite an unexpected panic on validation.

I couldn't yet reproduce it on a mock use-case, but it's 100% reproducible on our codebase.

Here's the stacktrace
```
runtime.gopanic
        /home/superstas/.gvm/gos/go1.22.6/src/runtime/panic.go:770
reflect.Value.Addr
        /home/superstas/.gvm/gos/go1.22.6/src/reflect/value.go:283
github.com/danielgtaylor/huma/v2.Register[...].func1.4
        /home/superstas/.gvm/pkgsets/go1.22.6/global/pkg/mod/github.com/danielgtaylor/huma/v2@v2.22.1/huma.go:1310
github.com/danielgtaylor/huma/v2.(*findResult[...]).everyPB
        /home/superstas/.gvm/pkgsets/go1.22.6/global/pkg/mod/github.com/danielgtaylor/huma/v2@v2.22.1/huma.go:358
github.com/danielgtaylor/huma/v2.(*findResult[...]).everyPB
        /home/superstas/.gvm/pkgsets/go1.22.6/global/pkg/mod/github.com/danielgtaylor/huma/v2@v2.22.1/huma.go:353
github.com/danielgtaylor/huma/v2.(*findResult[...]).everyPB
        /home/superstas/.gvm/pkgsets/go1.22.6/global/pkg/mod/github.com/danielgtaylor/huma/v2@v2.22.1/huma.go:336
github.com/danielgtaylor/huma/v2.(*findResult[...]).everyPB
        /home/superstas/.gvm/pkgsets/go1.22.6/global/pkg/mod/github.com/danielgtaylor/huma/v2@v2.22.1/huma.go:336
github.com/danielgtaylor/huma/v2.(*findResult[...]).everyPB
        /home/superstas/.gvm/pkgsets/go1.22.6/global/pkg/mod/github.com/danielgtaylor/huma/v2@v2.22.1/huma.go:336
github.com/danielgtaylor/huma/v2.(*findResult[...]).everyPB
        /home/superstas/.gvm/pkgsets/go1.22.6/global/pkg/mod/github.com/danielgtaylor/huma/v2@v2.22.1/huma.go:336
github.com/danielgtaylor/huma/v2.(*findResult[...]).EveryPB
        /home/superstas/.gvm/pkgsets/go1.22.6/global/pkg/mod/github.com/danielgtaylor/huma/v2@v2.22.1/huma.go:368
github.com/danielgtaylor/huma/v2.Register[...].func1
        /home/superstas/.gvm/pkgsets/go1.22.6/global/pkg/mod/github.com/danielgtaylor/huma/v2@v2.22.1/huma.go:1309
github.com/danielgtaylor/huma/v2/adapters/humaecho.(*echoAdapter).Handle.func1
        /home/superstas/.gvm/pkgsets/go1.22.6/global/pkg/mod/github.com/danielgtaylor/huma/v2@v2.22.1/adapters/humaecho/humaecho.go:123
github.com/labstack/echo/v4.(*Echo).add.func1
```

It looks like the fix is simple:

```diff
diff --git a/huma.go b/huma.go
index a5cafbb..7f431bb 100644
--- a/huma.go
+++ b/huma.go
@@ -1307,11 +1307,15 @@ func Register[I, O any](api API, op Operation, handler func(context.Context, *I)
 		}
 
 		resolvers.EveryPB(pb, v, func(item reflect.Value, _ bool) {
-			if resolver, ok := item.Addr().Interface().(Resolver); ok {
+			if item.CanAddr() {
+				item = item.Addr()
+			}
+
+			if resolver, ok := item.Interface().(Resolver); ok {
 				if errs := resolver.Resolve(ctx); len(errs) > 0 {
 					res.Errors = append(res.Errors, errs...)
 				}
-			} else if resolver, ok := item.Addr().Interface().(ResolverWithPath); ok {
+			} else if resolver, ok := item.Interface().(ResolverWithPath); ok {
 				if errs := resolver.Resolve(ctx, pb); len(errs) > 0 {
 					res.Errors = append(res.Errors, errs...)
 				}
```

But there are two open questions:
- I still can't reproduce it on a small isolated test-case ( so the root cause isn't clear ).
- Would this fix work well for non-addressable values ( tests are green, though )?

@danielgtaylor thank you for taking a look

Here's a test-case that reproduces this bug

main.go
```go
package main

import (
	"context"
	"net/http"

	"github.com/danielgtaylor/huma/v2"
	"github.com/danielgtaylor/huma/v2/adapters/humachi"
	"github.com/go-chi/chi/v5"
)

type (
	TagValue          string
	TagKey            string
	CustomHeader      string
	Tags              map[TagKey]TagValue
	UserCreateRequest struct {
		Name string `json:"name" minLength:"1" maxLength:"32"`
		Tags Tags   `json:"tags"`
	}

	Input struct {
		Body         UserCreateRequest
		CustomHeader CustomHeader `header:"X-Custom-Header" required:"true" minLength:"1" maxLength:"32"`
	}

	Output struct {
		Body struct {
			Message string `json:"message"`
		}
	}
)

func (k TagKey) TransformSchema(r huma.Registry, s *huma.Schema) *huma.Schema {
	min, max := 1, 32
	s.Type = "string"
	s.MinLength = &min
	s.MaxLength = &max
	return s
}

func (k TagKey) Resolve(ctx huma.Context, prefix *huma.PathBuffer) []error {
	if len(k) < 1 || len(k) > 32 {
		return []error{&huma.ErrorDetail{
			Location: prefix.String(),
			Message:  "Tag key must be between 1 and 32 characters",
			Value:    k,
		}}
	}

	return nil
}

func (v TagValue) TransformSchema(r huma.Registry, s *huma.Schema) *huma.Schema {
	min, max := 1, 128
	s.Type = "string"
	s.MinLength = &min
	s.MaxLength = &max
	return s
}

func (v TagValue) Resolve(ctx huma.Context, prefix *huma.PathBuffer) []error {
	if len(v) < 1 || len(v) > 128 {
		return []error{&huma.ErrorDetail{
			Location: prefix.String(),
			Message:  "Tag value must be between 1 and 128 characters",
			Value:    v,
		}}
	}

	return nil
}

func addRoutes(api huma.API) {
	huma.Post(api, "/user", func(ctx context.Context, input *Input) (*Output, error) {
		resp := &Output{}
		resp.Body.Message = "It works!"
		return resp, nil
	})
}

func main() {
	router := chi.NewMux()
	cfg := huma.DefaultConfig("My API", "1.0.0")
	api := humachi.New(router, cfg)

	addRoutes(api)

	http.ListenAndServe("127.0.0.1:8888", router)
}
```

main_test.go
```go
package main

import (
	"testing"

	"github.com/danielgtaylor/huma/v2"
	"github.com/danielgtaylor/huma/v2/humatest"
)

func TestUserCreate(t *testing.T) {
	_, api := humatest.New(t, huma.DefaultConfig("Test API", "1.0.0"))
	addRoutes(api)

	_ = api.Post("/user", "Header:", map[string]any{"name": "", "tags": map[string]any{"key": "value"}})

}
```

The result:
```
--- FAIL: TestUserCreate (0.00s)
    main_test.go:14: Making request:
        POST /user HTTP/1.1
        Content-Type: application/json
        Header: 
        
        {
          "name": "",
          "tags": {
            "key": "value"
          }
        }
panic: reflect.Value.Addr of unaddressable value [recovered]
	panic: reflect.Value.Addr of unaddressable value
```

I hope that helps.

--- a/huma.go
+++ b/huma.go
@@ -1307,11 +1307,24 @@ func Register[I, O any](api API, op Operation, handler func(context.Context, *I)
 		}
 
 		resolvers.EveryPB(pb, v, func(item reflect.Value, _ bool) {
-			if resolver, ok := item.Addr().Interface().(Resolver); ok {
+			if item.CanAddr() {
+				item = item.Addr()
+			} else {
+				// If the item is non-addressable (example: primitive custom type with
+				// a resolver as a map value), then we need to create a new pointer to
+				// the value to ensure the resolver can be called, regardless of whether
+				// is is a value or pointer resolver type.
+				// TODO: this is inefficient and could be improved in the future.
+				ptr := reflect.New(item.Type())
+				elem := ptr.Elem()
+				elem.Set(item)
+				item = ptr
+			}
+			if resolver, ok := item.Interface().(Resolver); ok {
 				if errs := resolver.Resolve(ctx); len(errs) > 0 {
 					res.Errors = append(res.Errors, errs...)
 				}
-			} else if resolver, ok := item.Addr().Interface().(ResolverWithPath); ok {
+			} else if resolver, ok := item.Interface().(ResolverWithPath); ok {
 				if errs := resolver.Resolve(ctx, pb); len(errs) > 0 {
 					res.Errors = append(res.Errors, errs...)
 				}

--- a/huma_test.go
+++ b/huma_test.go
@@ -2277,6 +2277,27 @@ func TestBodyRace(t *testing.T) {
 	}
 }
 
+type CustomMapValue string
+
+func (v *CustomMapValue) Resolve(ctx huma.Context) []error {
+	return nil
+}
+
+func TestResolverCustomTypePrimitive(t *testing.T) {
+	_, api := humatest.New(t, huma.DefaultConfig("Test API", "1.0.0"))
+	huma.Post(api, "/test", func(ctx context.Context, input *struct {
+		Body struct {
+			Tags map[string]CustomMapValue `json:"tags"`
+		}
+	}) (*struct{}, error) {
+		return nil, nil
+	})
+
+	assert.NotPanics(t, func() {
+		api.Post("/test", map[string]any{"tags": map[string]string{"foo": "bar"}})
+	})
+}
+
 // func BenchmarkSecondDecode(b *testing.B) {
 // 	//nolint: musttag
 // 	type MediumSized struct {

